samples/embedded-auth-with-sdk/src/main/java/com/okta/spring/example/controllers/LoginController.java

```diff
@@ -339,9 +337,26 @@ public ModelAndView register(final @RequestParam("lastname") String lastname,
         }
 
         UserProfile userProfile = new UserProfile();
-        userProfile.addAttribute("lastName", lastname);
-        userProfile.addAttribute("firstName", firstname);
-        userProfile.addAttribute("email", email);
+        FormValue userProfileFormValue = null;
+
+        for (FormValue formValue: newUserRegistrationResponse.getFormValues()) {
+            if (formValue.getName().contentEquals("userProfile")) {
+                userProfileFormValue = formValue;
+            }
+        }
+
+        if (userProfileFormValue == null) {
```

Comment:
Is there a better way to handle this error? Quite unlikely to happen though

Reply:
Streams API would simplify this

```java
        Optional<FormValue> userProfileFormValue =
                newUserRegistrationResponse.getFormValues()
                .stream().filter(x -> x.getName().equals("userProfile")).findFirst();

        if (!userProfileFormValue.isPresent()) {
```